File: chazm-model/src/main/java/runtimemodels/chazm/model/event/CapabilityEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.Capability;

/**
 * The {@linkplain CapabilityEventFactory} interface defines the API for constructing {@linkplain CapabilityEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface CapabilityEventFactory {

    /**
     * Constructs an {@linkplain CapabilityEvent}.
     *
     * @param category   the {@linkplain EventType}.
     * @param capability the {@linkplain Capability}.
     * @return a {@linkplain CapabilityEvent}.
     */
    CapabilityEvent build(EventType category, Capability capability);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/CapabilityEventFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Capability
import runtimemodels.chazm.model.event.EventType

/**
 * The [CapabilityEventFactory] interface defines the API for constructing [CapabilityEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface CapabilityEventFactory {
    /**
     * Constructs an [CapabilityEvent].
     *
     * @param category   the [EventType].
     * @param capability the [Capability].
     * @return a [CapabilityEvent].
     */
    fun build(category: EventType, capability: Capability): CapabilityEvent
}
